README.md

```diff
@@ -14,134 +20,432 @@ For suggestions on improvements,
+  ```
+  ```
+  location: https://d37ci6vzurychx.cloudfront.net/trip-data/yellow_tripdata_2022-01.parquet
+  ```
+- `format`: string. Required.  
+One of `parquet`, `spark_sql` and `delta`.  
 
-- Clean a step cache or all step caches:
-```
-mlflow pipelines clean --step step_name
-```
 
-**Note**: a short cut to `mlflow pipelines` is installed as `mlp`.
-For example, to run the ingest step,
-instead of issuing `mlflow pipelines run --step ingest`, you may type
+- `custom_loader_method`: string. Optional.  
+Fully qualified name of the custom loader function.  
+<u>Example</u>: 
+  ```
+  custom_loader_method: steps.ingest.load_file_as_dataframe
+  ```
+
+- `sql`: string. Required if format is `spark_sql`.  
+Specifies a SparkSQL statement that identifies the dataset to use.
+
+
+- `version`: int. Optional.  
+If the `delta` format is specified, use this to specify the Delta table version to read from.
+
+
+- `timestamp`: timestamp. Optional.  
+If the `delta` format is specified, use this to specify the timestamp at which to read data.
+</details>
+
+**Step artifacts**
+- `ingested_data`: The ingested data as a Pandas DataFrame.
+
+### Split step
+
+The split step splits the ingested dataset produced by the ingest step into:
+- a training dataset for model training
+- a validation dataset for model performance evaluation & tuning, and 
+- a test dataset for model performance evaluation.  
+
+The fraction of records allocated to each dataset is defined by the `split_ratios` attribute of the `split` step
+definition in [`pipeline.yaml`](https://github.com/mlflow/mlp-regression-template/blob/main/pipeline.yaml). 
+The split step also preprocesses the datasets using logic defined in [`steps/split.py`](https://github.com/mlflow/mlp-regression-template/blob/main/steps/split.py).
+Subsequent steps use these datasets to develop a model and measure its performance.
+
+The post-split method should be written in `steps/split.py` and should accept three parameters:
+- `train_df`: DataFrame. The unprocessed train dataset.
+- `validation_df`: DataFrame. The unprocessed validation dataset.
+- `test_df`: DataFrame. The unprocessed test dataset.
+
+It should return a triple representing the processed train, validation and test datasets. `steps/split.py` contains an example placeholder function.
+
+The split step is configured by the `steps.split` section in `pipeline.yaml` as follows:
+<details>
+<summary>Reference</summary>
+
+- `split_ratios`: list. Optional.  
+A YAML list specifying the ratios by which to split the dataset into training, validation and test sets.  
+<u>Example</u>: 
+  ```
+  split_ratios: [0.75, 0.125, 0.125] # Defaults to this ratio if unspecified
+  ```
+- `post_split_method`: string. Optional.   
+Fully qualified name of the method to use to "post-process" the split datasets. 
+This procedure is meant for removing/filtering records, or other cleaning processes. Arbitrary data transformations 
+should be done in the transform step.  
```

Comment:
Can we update this to `post_split_filter_method` to match the changes in https://github.com/mlflow/mlflow/pull/6928?